opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -718,6 +705,78 @@ namespace Opm
 
 
 
+    template<typename TypeTag>
+    bool
+    WellInterface<TypeTag>::
+    checkMaxRatioLimitWell(const WellState& well_state,
+                           const double max_ratio_limit,
+                           double (*ratioFunc)(const std::vector<double>&, const PhaseUsage&)) const
```

Comment:
No need to require function pointers here, it is hard to read and prevents inlining. Rather, make this a template member taking a RatioFunctor template argument. The calling code should not require changes.

Reply:
Please check it is what you want. I do not use it very often.

Reply:
Updated with more communications.